[BUG] Opennext build cannot run in Docker

### Describe the bug

When trying to run `npx opennextjs-cloudflare build` (or `npx @opennextjs/cloudflare build`) in Docker (using image `public.ecr.aws/docker/library/node:20.19.4-slim`)

The issue occurs on version `1.14.6`, using `1.12.0` works just fine, so it is probably a regression introduced recently (seems like it was introduced in `1.13.0`)

Stack trace:
```
Error: Cannot find native binding. npm has a bug related to optional dependencies (https://github.com/npm/cli/issues/4828). Please try `npm i` again after removing both package-lock.json and node_modules directory.
    at Object.<anonymous> (/app/workdir/medusajs-cloud-fullstack-starter-b39d0352c84d513e4dfb3d29c6b550f8f7a0b5ce/node_modules/.pnpm/@ast-grep+napi@0.40.0/node_modules/@ast-grep/napi/index.js:385:11)
    at Module._compile (node:internal/modules/cjs/loader:1529:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1613:10)
    at Module.load (node:internal/modules/cjs/loader:1275:32)
    at Module._load (node:internal/modules/cjs/loader:1096:12)
    at cjsLoader (node:internal/modules/esm/translators:298:15)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:240:7)
    at ModuleJob.run (node:internal/modules/esm/module_job:263:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:540:24)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:117:5) {
  [cause]: [
    Error: Cannot find module './ast-grep-napi.linux-arm64-gnu.node'
    Require stack:
    - /app/workdir/medusajs-cloud-fullstack-starter-b39d0352c84d513e4dfb3d29c6b550f8f7a0b5ce/node_modules/.pnpm/@ast-grep+napi@0.40.0/node_modules/@ast-grep/napi/index.js
        at Module._resolveFilename (node:internal/modules/cjs/loader:1212:15)
        at Module._load (node:internal/modules/cjs/loader:1043:27)
        at Module.require (node:internal/modules/cjs/loader:1298:19)
        at require (node:internal/modules/helpers:182:18)
        at requireNative (/app/workdir/medusajs-cloud-fullstack-starter-b39d0352c84d513e4dfb3d29c6b550f8f7a0b5ce/node_modules/.pnpm/@ast-grep+napi@0.40.0/node_modules/@ast-grep/napi/index.js:237:18)
        at Object.<anonymous> (/app/workdir/medusajs-cloud-fullstack-starter-b39d0352c84d513e4dfb3d29c6b550f8f7a0b5ce/node_modules/.pnpm/@ast-grep+napi@0.40.0/node_modules/@ast-grep/napi/index.js:362:17)
        at Module._compile (node:internal/modules/cjs/loader:1529:14)
        at Module._extensions..js (node:internal/modules/cjs/loader:1613:10)
        at Module.load (node:internal/modules/cjs/loader:1275:32)
        at Module._load (node:internal/modules/cjs/loader:1096:12) {
      code: 'MODULE_NOT_FOUND',
      requireStack: [
        '/app/workdir/medusajs-cloud-fullstack-starter-b39d0352c84d513e4dfb3d29c6b550f8f7a0b5ce/node_modules/.pnpm/@ast-grep+napi@0.40.0/node_modules/@ast-grep/napi/index.js'
      ]
    },
    Error: /app/workdir/medusajs-cloud-fullstack-starter-b39d0352c84d513e4dfb3d29c6b550f8f7a0b5ce/node_modules/.pnpm/@ast-grep+napi-linux-arm64-gnu@0.40.0/node_modules/@ast-grep/napi-linux-arm64-gnu/ast-grep-napi.linux-arm64-gnu.node: undefined symbol: le16toh
        at Module._extensions..node (node:internal/modules/cjs/loader:1651:18)
        at Module.load (node:internal/modules/cjs/loader:1275:32)
        at Module._load (node:internal/modules/cjs/loader:1096:12)
        at Module.require (node:internal/modules/cjs/loader:1298:19)
        at require (node:internal/modules/helpers:182:18)
        at requireNative (/app/workdir/medusajs-cloud-fullstack-starter-b39d0352c84d513e4dfb3d29c6b550f8f7a0b5ce/node_modules/.pnpm/@ast-grep+napi@0.40.0/node_modules/@ast-grep/napi/index.js:242:18)
        at Object.<anonymous> (/app/workdir/medusajs-cloud-fullstack-starter-b39d0352c84d513e4dfb3d29c6b550f8f7a0b5ce/node_modules/.pnpm/@ast-grep+napi@0.40.0/node_modules/@ast-grep/napi/index.js:362:17)
        at Module._compile (node:internal/modules/cjs/loader:1529:14)
        at Module._extensions..js (node:internal/modules/cjs/loader:1613:10)
        at Module.load (node:internal/modules/cjs/loader:1275:32) {
      code: 'ERR_DLOPEN_FAILED'
    }
  ]
}

Node.js v20.19.4
```

### Steps to reproduce

(This is the minimum reproduction, our Dockerfile is more complex, but the issue is the same)

1. Create a very basic Dockerfile:
```
FROM public.ecr.aws/docker/library/node:20.19.4-slim AS base
RUN npx @opennextjs/cloudflare build
```
2. Run `docker build .`
3. See the error

### Expected behavior

opennext cloudflare works as expected n the latest version

### @opennextjs/cloudflare version

1.14.6

### Wrangler version

4.55

### next info output

```shell
N/A
```

### Additional context

_No response_

--- a/packages/cloudflare/package.json
+++ b/packages/cloudflare/package.json
@@ -52,7 +52,7 @@
 	},
 	"homepage": "https://github.com/opennextjs/opennextjs-cloudflare",
 	"dependencies": {
-		"@ast-grep/napi": "0.40.0",
+		"@ast-grep/napi": "^0.40.5",
 		"@dotenvx/dotenvx": "catalog:",
 		"@opennextjs/aws": "3.9.14",
 		"cloudflare": "^4.4.1",

